castxml cannot find stddef.h under nix-shell

I am trying to generate Python bindings for [NS-3](https://www.nsnam.org/) in the `nix-shell` environment. It uses [pybindgen](https://github.com/gjcarneiro/pybindgen) and `castxml` for that.



I have constructed a minimal example:
```
#!/usr/bin/env nix-shell
#!nix-shell -i python -p castxml pythonPackages.pygccxml
from pygccxml import utils, parser

generator_path, generator_name = utils.find_xml_generator()

xml_generator_config = parser.xml_generator_configuration_t(
    xml_generator_path=generator_path,
    xml_generator=generator_name)

parser.parse_string("#include <wchar.h>", xml_generator_config)
```

I expect this code to exit without errors as it does on my Debian system:
```
$ pip install --user pybindgen
# apt-get install castxml
$ python example
$ # no errors
```

This is how it fails under `nix-shell`:
```
$ chmod +x example
$ ./example
warning: argument unused during compilation: '-nostdlibinc' [-Wunused-command-line-argument]
In file included from /run/user/1000/tmpdUub09.h:1:
/nix/store/bniand9afisrgrsfi7kr093334iv3ibv-glibc-2.27-dev/include/wchar.h:35:10: fatal error: 'stddef.h' file not found
#include <stddef.h>
         ^~~~~~~~~~
1 error generated.
Traceback (most recent call last):
  File "./example", line 11, in <module>
    parser.parse_string("#include <wchar.h>", xml_generator_config)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/__init__.py", line 60, in parse_string
    declarations = parser.read_string(content)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/project_reader.py", line 369, in read_string
    decls = reader.read_string(content)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/source_reader.py", line 430, in read_string
    decls = self.read_file(header_file)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/source_reader.py", line 356, in read_file
    return self.read_cpp_source_file(source_file)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/source_reader.py", line 375, in read_cpp_source_file
    xml_file = self.create_xml_file(ffname)
  File "/nix/store/yc25lx6hpnfphll9adscwznxv1yvxbf9-python2.7-pygccxml-1.9.1/lib/python2.7/site-packages/pygccxml/parser/source_reader.py", line 324, in create_xml_file
    ": %s status:%s" % (gccxml_msg, exit_status))
RuntimeError: Error occurred while running CASTXML:  status:1
```

This problem was discussed before in `#nixos` channel: https://logs.nix.samueldr.com/nixos/2018-06-01#1259324
Related NS-3 PR by @teto stuck with the same problem: https://github.com/NixOS/nixpkgs/pull/41264
Merge request on GitLab where I try to build a `nix-shell` environment for Python bindings generation: https://gitlab.com/nsnam/ns-3-dev/merge_requests/88

The bug is not even about `pygccxml`, I have reduced it to `castxml` only:
```
#!/usr/bin/env nix-shell
#!nix-shell -i sh -p castxml
cat >example.cc <<EOF
#include <wchar.h>
EOF
castxml
castxml --castxml-cc-gnu gcc example.cc
```

Output:
```
$ ./example
warning: argument unused during compilation: '-nostdlibinc' [-Wunused-command-line-argument]
In file included from example.cc:1:
/nix/store/bniand9afisrgrsfi7kr093334iv3ibv-glibc-2.27-dev/include/wchar.h:35:10: fatal error: 'stddef.h' file not found
#include <stddef.h>
         ^~~~~~~~~~
1 error generated.
```

Save this to `1.c`:
```
#include <stddef.h>
```

Run `nix-shell -p clang castxml` and compile the file with `clang`:
```
$ clang -c 1.c -v
...
clang -cc1 version 7.1.0 based upon LLVM 7.1.0 default target x86_64-unknown-linux-gnu
ignoring duplicate directory "/nix/store/bsdig6vdnmlxmvmqwg0hfvp3hvhb9myi-compiler-rt-7.1.0-dev/include"
ignoring duplicate directory "/nix/store/bsdig6vdnmlxmvmqwg0hfvp3hvhb9myi-compiler-rt-7.1.0-dev/include"
#include "..." search starts here:
#include <...> search starts here:
 /nix/store/bsdig6vdnmlxmvmqwg0hfvp3hvhb9myi-compiler-rt-7.1.0-dev/include
 /nix/store/q82rffgvpsr3j1v009hf75yj2lhy91fl-clang-wrapper-7.1.0/resource-root/include
 /nix/store/bniand9afisrgrsfi7kr093334iv3ibv-glibc-2.27-dev/include
End of search list.
```

The file `/nix/store/q82rffgvpsr3j1v009hf75yj2lhy91fl-clang-wrapper-7.1.0/resource-root/include/stddef.h` is in the search list.

Run `castxml`:
```
$ castxml -v 1.c
...
clang -cc1 version 7.1.0 based upon LLVM 7.1.0 default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /nix/store/k0dda3b4z4hz4vvfbid7lqpy7c91qjc1-CastXML-0.2.0/share/castxml/clang/include
End of search list.
1.c:1:10: fatal error: 'stddef.h' file not found
#include <stddef.h>
         ^~~~~~~~~~
1 error generated.
```

The command
```
find /nix/store/k0dda3b4z4hz4vvfbid7lqpy7c91qjc1-CastXML-0.2.0/share/castxml/clang/include/ | grep -c stddef
```
outputs "0".

In Debian, CastXML package contains `/usr/share/castxml/clang/include/stddef.h`: https://packages.debian.org/sid/amd64/castxml/filelist
Debian has CastXML version `0.1+git20180702-3` according to https://packages.debian.org/sid/castxml

--- a/pkgs/development/tools/castxml/default.nix
+++ b/pkgs/development/tools/castxml/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub
+{ lib, stdenv, fetchFromGitHub
 , pythonPackages
 , cmake
 , llvmPackages
@@ -19,8 +19,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ] ++ stdenv.lib.optionals withMan [ pythonPackages.sphinx ];
 
+  clangVersion = lib.getVersion llvmPackages.clang;
+
   cmakeFlags = [
-    "-DCLANG_RESOURCE_DIR=${llvmPackages.clang-unwrapped}"
+    "-DCLANG_RESOURCE_DIR=${llvmPackages.clang-unwrapped}/lib/clang/${clangVersion}/"
     "-DSPHINX_MAN=${if withMan then "ON" else "OFF"}"
   ];
 

